### ISSUE TITLE
RFE: Display target number for MASClike systems next to movement.

### ISSUE THREAD

**Issue by MarcJC** [MegaMek/megamek#3418]
I am on 49.6 if it matters.  

When using MASC or Supercharger a roll is needed to activate the system successfully.  
It would be nice if on the Unit Display window - on the General tab, if the movement section that could be altered by rolling for activating a system would display the target number on that screen.  

So in this it would show 

Walk 4 
Run 6(8) - (5) 

That would let me know that if I want to use my MASC I will have to roll over a 5 to be successful.   

This RFE should display for any MASC or MASC like system that works this way.  I know Superchargers also work in this way.  I am unsure if there is any other equipment that would be effected.  I know VTOLs have a jet booster but I am unsure of the rules on that system, it is something that should be looked up.  

The display would not be used for moving faster than movement profile on something that simply requires a piloting check, such as movement in low gravity or the like.    

I was informed on the Discord that uncoupling the MASC from the Supercharger is currently being done.  
This will cause an edge case where a unit has two (or in the future possibly more) systems which each have their own target number.  What to do in that case is left to best GUI principles, however my advice would be to simply display the lowest number of the current existing active systems.  

Which brings in another edge case - If the system is broken, from failing an activation roll or taking a critical hit, I would expect not to display the target number as the unit can not move that fast.  However it looks to be possible for MASC to be deactivated by the mode button (I'm not sure why someone would do this).  In this case the question is if the target number of a working system which is deactivated but can be turned back on should be shown.  I leave that question to the designers.  

In addition I don't know that all people would want to display this information or not.  I would think allowing the information to be toggled on or off by a check box in the client settings options would be appropriate.   

Thank you

**Comment by pakfront** [comment]
I'll take a look at this in conjunction with https://github.com/MegaMek/megamek/issues/2749

**Comment by pakfront** [comment]
FYI - you might want to 'disarm' masc or supercharger if you have both.  The chance of a crit fail goes up each turn you use them. If you switch which one you use each turn  you can keep the critical failure chance low. At least, this will be true once we decouple the two.

**Comment by MarcJC** [comment]
what would be the expected behavior if you have both systems and they are both armed but you only use up to double your movement and not 2.5 times your movement which would be both systems, after the expected upgrade goes in?  

It would seem like if they had different numbers the one with the better target number should be chosen, but that runs into a check with the enhanced better MASC option where you can be higher up on the track but have the same target number as a previous track target.  I don't know how the target numbers are stored but I would guess they are initialized as a constant of an array of INTs at the global level, for easy look up.  Then each mech would simply need it's own tracker for where each of it's equipment's were on the track.  Obviously simply picking the the array offset that is lowest should work for which one to use.    

If both numbers are the same then you do have a question of which to use.  My thought would be to choose the supercharger over the MASC as the penalties for failure are generally lower.  However some people might want to default to the other way so a checkbox option might be something called for to allow the player to select for themselves.  I can imagine that if more equipment was added this could become a more complex calculation so probably going to want to leave a comment flag there to know where to get back to if more equipment is added.

**Comment by pakfront** [comment]
Yes, that is the plan - see https://github.com/MegaMek/megamek/issues/2749
